src/common/kube-helpers.ts

```diff
@@ -140,3 +141,23 @@ export function getNodeWarningConditions(node: V1Node) {
     c.status.toLowerCase() === "true" && c.type !== "Ready" && c.type !== "HostUpgrades"
   )
 }
+
+export function validateKubeConfig(config: KubeConfig) {
+  logger.debug(`validateUserConfig: config received - ${JSON.stringify(config)}`);
+  // we only receive a single context, cluster & user object here so lets validate them as this
+  // will be called when we add a new cluster to Lens
+
+  // Validate the User Object
+  const user = config.getCurrentUser();
+  if (user.exec) {
+    const execCommand = user.exec["command"]; 
+    // validate the exec struct in the user object, start with the command field
+    logger.debug(`validateUserConfig: validating user exec command - ${JSON.stringify(execCommand)}`);
+    if (!commandExistsSync(execCommand)) {
+      logger.debug(`validateUserConfig: exec command ${String(execCommand)} in kubeconfig ${config.currentContext} not found`);
+      return `User Exec command "${execCommand}" not found on host.`;
```

Comment:
How about throwing error here?

Reply:
@nevalla - What I was struggling with is if I throw an error here then it will break the try/catch block in add-clusters and basically stop the import of any other contexts. Is that what we want?

Reply:
You could always add another `try/catch` and rethrow if not an instance of `ExecValidationError`

Reply:
Done by adding custom error as suggested by @Nokel81 and catching it higher up.